### PR TITLE
Update Scopes instead of creating new ones if code is the same

### PR DESCRIPTION
**Pull request by tramuntanal**
#### :tophat: What? Why?
When updating scopes from yaml files if the code was the same but the name changed a new scope was created.
This PR resolves this situation and simply updates the Scope instead of resulting in two scopes with the same code and different names.

### CODE REVIEW

File: lib/tasks/civi_crm.rake [hunk 1]
```diff
@@ -63,6 +65,79 @@ namespace :civi_crm do
     end
   end
 
+  namespace :import_by_csv do
```
Comment (tramuntanal) [inline lib/tasks/civi_crm.rake]: can we better namespace the CSV related tasks please?
```suggestion
  namespace :erc_auth do
    namespace :csv_import do
      namespace :comarcals do
      ....
```

Also, can we extract this into a specific csv.rake file?

File: lib/tasks/civi_crm.rake [hunk 2]
```diff
@@ -63,6 +65,79 @@ namespace :civi_crm do
...
+
+    def get_code_relationships(local_ids, _filter_ids, scope_type)
+      local_ids.each_with_object({}) do |element, memo|
+        csv_text = File.read("tmp/scopes_hierarchy.csv")
+        csv = CSV.parse(csv_text, headers: true)
+
+        csv = csv.select { |row| row["SL"] == element }.first.to_h
+        next if csv.blank?
+
+        memo[csv["SL"]] = csv["FC"] if scope_type == "comarcal"
```
Comment (tramuntanal) [inline lib/tasks/civi_crm.rake]: case .. when..?

**Review by tramuntanal — COMMENTED** [review]
LGTM @LauraJaimeC just a couple of details :smile:

File: lib/file_manager.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module FileManager
```
Comment (tramuntanal) [inline lib/file_manager.rb:3]: :heart:

File: lib/tasks/erc_auth.rake [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+      csv = csv.select { |row| row["type"] == "SL" }
+      csv = csv.map(&:to_h)
+
+      result = get_code_and_display_name(csv)
+      write_config_yaml!("locals", result)
+    end
+
+    desc "Generates a YAML file with the relationship between CSV 'Local' and 'Comarcal'"
+    task local_comarcal_relationships: :environment do
```
Comment (tramuntanal) [inline lib/tasks/erc_auth.rake:49]: isn't this task file the same as in `civi_crm`?

File: lib/tasks/erc_auth.rake [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      local_ids = load_config_yaml("locals").keys
+      comarcal_ids = load_config_yaml("comarcals").keys
+      result = get_code_relationships(local_ids, comarcal_ids, "comarcal")
+      write_config_yaml!("local_comarcal_relationships", result)
+    end
+
+    desc "Generates a YAML file with the relationship between CSV 'Local' and 'Regional'"
+    task local_regional_relationships: :environment do
```
Comment (tramuntanal) [inline lib/tasks/erc_auth.rake:57]: isn't this task file the same as in civi_crm?

File: lib/tasks/erc_auth.rake (hunk 4 above)
Comment (laurajaime) [reply lib/tasks/erc_auth.rake:49]: I not the same, look:

in cvm_crm:

**data = get_civi_crm_data(:find_local_organization_relationships)**
comarcal_ids = load_config_yaml("comarcals").keys
**result = get_contact_id_relationships(data, comarcal_ids)**
write_config_yaml!("local_comarcal_relationships", result)

in erc_auth:

**local_ids = load_config_yaml("locals").keys**
comarcal_ids = load_config_yaml("comarcals").keys
**result = get_code_relationships(local_ids, comarcal_ids, "comarcal")**
write_config_yaml!("local_comarcal_relationships", result)

The same happens in local_regional_relationships

**Pull request merged**